compiler/noirc_evaluator/src/ssa/ir/types.rs

```diff
@@ -29,15 +31,28 @@ impl NumericType {
         }
     }
 
-    /// Returns true if the given Field value is within the numeric limits
+    /// Returns true if the given BigInt value is within the numeric limits
     /// for the current NumericType.
-    pub(crate) fn value_is_within_limits(self, field: FieldElement) -> bool {
+    pub(crate) fn value_is_within_limits(self, value: &BigInt) -> bool {
         match self {
-            NumericType::Signed { bit_size } | NumericType::Unsigned { bit_size } => {
-                let max = 2u128.pow(bit_size) - 1;
-                field <= max.into()
+            NumericType::Signed { bit_size } => {
+                let min: BigInt = -BigInt::from_i128(1_i128 << (bit_size - 1)).unwrap();
+                let max: BigInt = BigInt::from_i128(1_i128 << (bit_size - 1)).unwrap() - 1;
+                min <= *value && *value <= max
+            }
+            NumericType::Unsigned { bit_size } => {
+                !value.is_negative() && value.bits() <= bit_size.into()
+            }
+            NumericType::NativeField => {
+                let modulus = FieldElement::modulus();
+                let modulus_big_int = BigInt::from_biguint(Sign::Plus, modulus);
```

Comment:
Same here, I wonder if there's a way to avoid creating this `BigInt`.

Reply:
What's the way to declare global variables in Rust?

Reply:
`const MODULUS:  BigInt = blah`

Reply:
Thanks!

I tried it:

```rust
    const MODULUS: BigInt = BigInt::from_biguint(Sign::Plus, FieldElement::modulus());
```

but I get this error:

```
cannot call non-const fn `num_bigint::BigInt::from_biguint` in constants
calls in constants are limited to constant functions, tuple structs and tuple variants
```